Engine/ResourceManagement/Manager/TextureManager.cpp

```diff
@@ -53,9 +66,65 @@ std::shared_ptr<Texture> TextureManager::Load(uint32_t uuid, const FileData& res
 	{
 		loaded_texture = std::make_shared<Texture>(uuid, data.data(), data.size(), width, height, num_channels, texture_options);
 	}
+	APP_LOG_SUCCESS("Time Loading Texture Manager: %.3f", timer.Pause());
+
 	return loaded_texture;
 }
 
+std::shared_ptr<Texture> TextureManager::LoadThread(uint32_t uuid, const FileData& resource_data, TextureLoadData& texture_data)
+{
+	if (timer.IsPaused())
+	{
+		timer.Resume();
+	}
+	else
+	{
+		timer.Start();
+	}
+	BROFILER_CATEGORY("Load Texture Manager", Profiler::Color::PaleGoldenRod);
+	std::string extension;
+	TextureOptions texture_options;
+	char* cursor = (char*)resource_data.buffer;
+	size_t bytes = sizeof(TextureOptions);
+	memcpy(&texture_options, cursor, bytes);
+
+	cursor += bytes;
+	bytes = extension_size;
+	extension.resize(extension_size);
+	memcpy(extension.data(), cursor, bytes);
+
+	size_t offset = extension_size + sizeof(TextureOptions);
+
+	std::vector<char> data;
+	int width, height, num_channels = 0;
+	bool normal_map = texture_options.texture_type == TextureType::NORMAL;
+	if (normal_map)
+	{
+		data = LoadImageData(resource_data, offset, extension, width, height, num_channels);
+	}
+	else
+	{
+		DDS::DDS_HEADER ddsHeader;
+		data = LoadCompressedDDS(resource_data, offset, ddsHeader);
+		width = ddsHeader.dwWidth;
+		height = ddsHeader.dwHeight;
+	}
+
+	if (data.size())
```

Comment:
This parts from above is the same as the usual Load, maybe you can merge both or create a common method

Reply:
Sure!